tests.py

```diff
@@ -47,3 +47,6 @@ def test_request_with_body(self):
     def test_request_with_https(self):
         response = self.session.get('https://localhost/index')
         self.assertEqual(response.json()['server_port'], 443)
+
+    def test_request_with_json(self):
+        response = self.session.post('http://localhost/index', json={})
```

Comment:
I've added some asserts mimicking a similar test- I don't know if it makes sense // follows your testing style.